### PR TITLE
For #2346. Enable kotlin warningsAsErrors for `feature-prompts` module.

### PR DESCRIPTION
**Pull request by dector**
### Issue #2346

Trivial changes in tests.

### Complexity

Easy (★☆☆)

### Pull Request checklist
- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] ~~**Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one~~
- [x] ~~**Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features~~

### CODE REVIEW

File: components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt [hunk 1]
```diff
@@ -149,6 +149,7 @@ class PromptFeature(
      * @param grantResults The grant results for the corresponding permissions
      * @see [onNeedToRequestPermissions].
      */
+    @Suppress("UNUSED_PARAMETER")
```
Comment (dector) [inline components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt]: `permissions` is unused. Seems like misleading implementation. Should I raise issue on it?

File: components/feature/prompts/src/main/java/mozilla/components/feature/prompts/TimePickerDialogFragment.kt [hunk 2]
```diff
@@ -110,7 +110,7 @@ internal class TimePickerDialogFragment : PromptDialogFragment(), DatePicker.OnD
         return view
     }
 
-    @SuppressWarnings("deprecation")
+    @Suppress("DEPRECATION")
```
Comment (dector) [inline components/feature/prompts/src/main/java/mozilla/components/feature/prompts/TimePickerDialogFragment.kt]: There is no way to set upper bound for required API, unfortunately.

File: components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt (hunk 1 above)
Comment (NotWoods) [reply components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt]: I believe the intent is to have a similar function signature to [`onRequestPermissionsResult`](https://developer.android.com/reference/android/support/v4/app/ActivityCompat.OnRequestPermissionsResultCallback).

Comment (dector) [reply components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt]: I think you are right. However parameter that is present in public API but absent in actual implementation can be potential cause of problems in the future. :)

**Pull request merged**